zebrad/tests/acceptance.rs

```diff
@@ -351,3 +354,69 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn listener_address_test1() -> Result<()> {
+    listener_address(
+        "localhost:0",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:"),
+    )?;
+    Ok(())
+}
+#[test]
+fn listener_address_test2() -> Result<()> {
+    listener_address(
+        "127.0.0.1:0",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:"),
+    )?;
+    Ok(())
+}
+#[test]
+fn listener_address_test3() -> Result<()> {
+    listener_address(
+        "::1:0",
```

Comment:
The IPv6 address must be separated from the port, using square brackets.

Otherwise, the parse is ambiguous. Some resolvers will try to resolve these addresses, and time out.

```suggestion
        "[::1]:0",
```